src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2403,8 +2404,28 @@ def aks_mesh_enable(
         client,
         resource_group_name,
         name,
+        key_vault_id=None,
+        ca_cert_object_name=None,
+        ca_key_object_name=None,
+        root_cert_object_name=None,
+        cert_chain_object_name=None
 ):
-    return _aks_mesh_update(cmd, client, resource_group_name, name, enable_azure_service_mesh=True)
+    instance = client.get(resource_group_name, name)
+    addon_profiles = instance.addon_profiles
+    if key_vault_id is not None and ca_cert_object_name is not None and ca_key_object_name is not None and root_cert_object_name is not None and cert_chain_object_name is not None:
+        if not addon_profiles or not addon_profiles[CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME] or not addon_profiles[CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME].enabled:
+            raise CLIError('AzureKeyvaultSecretsProvider addon is required for Azure Service Mesh plugin certificate authority feature.')
```

Comment:
may take some error type like RequiredArgumentMissingError

Reply:
I see CLIError being used for similar errors. RequiredArgumentMissingError is probably not right since AzureKeyvaultSecretsProvider  is another addon that is required to be enabled before this feature can be used.
  
https://github.com/Azure/azure-cli-extensions/blob/ea718b516a2371c2ddcc1a9f245ac87733978f2a/src/aks-preview/azext_aks_preview/custom.py#L1817C12-L1830C64